x-pack/plugins/screenshotting/server/screenshots/index.ts

```diff
@@ -232,15 +217,20 @@ export class Screenshots {
   getScreenshots(options: PdfScreenshotOptions): Observable<PdfScreenshotResult>;
   getScreenshots(options: ScreenshotOptions): Observable<ScreenshotResult>;
   getScreenshots(options: ScreenshotOptions): Observable<ScreenshotResult> {
-    const transaction = apm.startTransaction('screenshot-pipeline', 'screenshotting');
-    const layout = this.createLayout(transaction, options);
+    const eventLogger = new EventLogger(this.logger, this.config);
+    eventLogger.screenshottingStart();
+
+    const layout = this.createLayout(options);
     const captureOptions = this.getCaptureOptions(options);
 
-    return this.captureScreenshots(layout, transaction, captureOptions).pipe(
+    return this.captureScreenshots(eventLogger, layout, captureOptions).pipe(
+      tap((results) => {
+        eventLogger.screenshottingEnd(results);
+      }),
```

Comment:
Shall the "screenshotting" capture also include `mergeMap` below, or, if not, be inside the `this.captureScrenshots` method?

Reply:
I looked through again and still think the code is laid out correctly.

There should be a divide between two transactions: the `screenshotting` and `generatePdf` transactions. The "generatePdf" transaction begins inside of the `pngsToPdf` function in `pdf_maker`.

The allows the duration of the `screenshotting` action to exclude the time it takes to compile the PDF.